src/OrleansRuntime/Storage/MemoryStorageGrain.cs

```diff
@@ -109,7 +111,7 @@ public void DeleteGrainState(string grainId, string receivedEtag)
                 }
 
                 ValidateEtag(currentETag, receivedEtag, grainId, "Delete");
-                grainStateStorage.Remove(grainId);
+                grainStateStorage[grainId] = Deleted;
```

Comment:
please also remove the extra TryGetValue we found in https://github.com/dotnet/orleans/pull/1885/files#diff-092591129736f7c70176968a95526331L105